imagesgallery/imagesgallery.py

```diff
@@ -263,6 +261,7 @@ def update_contents(self, content):
     @XBlock.json_handler
     def get_files(self, data, suffix=''):  # pylint: disable=unused-argument
         """Handler for getting images from the course assets."""
+        self.sync_course_assets()
```

Comment:
You need to call this only when in the edit studio view where we can edit the xblock state. You can use this check: https://github.com/eduNEXT/xblock-filesmanager/pull/17/files#diff-34c28897396ed287b41be425c22317e68f699f1faab8d7c6e4c31f21eb1c5b32L296-L297
Since in the edit view there's no anonymous ID context.

Reply:
Thanks! I made the changes